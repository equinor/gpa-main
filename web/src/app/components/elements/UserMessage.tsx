import React from "react";
import styled from "styled-components";
import { EquinorIcon, EIcon } from "../../assets/svg/EquinorIcon";
import { EColor } from "../../common/Color";

interface IUserMessage {
    type: "loading" | "error"
    text: string;
}

export const UserMessage = (props: IUserMessage) => {
    return (
        <StUserMessage type={props.type}>
            {props.type === "loading" &&
                <EquinorIcon icon={EIcon.LOADING} size={24} />
            }
            {props.type === "error" &&
                <EquinorIcon icon={EIcon.ERROR} size={24} />
            }
            <span>{props.text}</span>
        </StUserMessage>
    )
}

interface IStUserMessage {
    type?: "loading" | "error";
}

const StUserMessage = styled.div`
    display: flex;
    width: 250px;
    color: ${(props: IStUserMessage) => (props.type === "loading" ? EColor.GREEN : EColor.DANGER_RED)};
    background-color: ${(props: IStUserMessage) => (props.type === "loading" ? EColor.LIGHT_GREEN : EColor.DANGER_LIGHT_RED)};
    height: 100px;
    border-radius: 5px;
    flex-direction: column;
    align-items: center; 
    > svg {
        -webkit-animation: ${(props: IStUserMessage) => (props.type === "loading" ? "spin 2s linear infinite" : "")};
        -moz-animation: ${(props: IStUserMessage) => (props.type === "loading" ? "spin 2s linear infinite" : "")};
        animation: ${(props: IStUserMessage) => (props.type === "loading" ? "spin 2s linear infinite" : "")};
        margin: 20px 0 18px 0;
    }
    > span {
        font-weight: bold;
    }
    @-moz-keyframes spin { 100% { -moz-transform: rotate(360deg); } }
    @-webkit-keyframes spin { 100% { -webkit-transform: rotate(360deg); } }
    @keyframes spin { 100% { -webkit-transform: rotate(360deg); transform:rotate(360deg); } }
`;