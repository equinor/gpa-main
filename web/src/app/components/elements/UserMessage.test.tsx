import { MockedProvider } from '@apollo/react-testing';
import React from 'react';
import renderer from 'react-test-renderer';

import { UserMessage } from './UserMessage';

describe("UserMessage", () => {
    it("snapshots", () => {
        const tree = renderer.create(
            <MockedProvider addTypename={false}>
                <UserMessage text={"text"} type={"loading"} />
            </MockedProvider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
}); 