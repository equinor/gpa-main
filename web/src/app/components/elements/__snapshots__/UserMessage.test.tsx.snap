// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserMessage snapshots 1`] = `
<div
  className="sc-bdVaJa cqSazf"
  type="loading"
>
  <svg
    fill="none"
    height={24}
    viewBox="0 0 24 24"
    width={24}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M12 1.84615C6.39219 1.84615 1.84615 6.39219 1.84615 12C1.84615 17.6078 6.39219 22.1538 12 22.1538C17.6078 22.1538 22.1538 17.6078 22.1538 12C22.1538 6.39219 17.6078 1.84615 12 1.84615ZM0 12C0 5.37258 5.37258 0 12 0C18.6274 0 24 5.37258 24 12C24 18.6274 18.6274 24 12 24C5.37258 24 0 18.6274 0 12Z"
      fill="#DEEDEE"
      fillRule="evenodd"
    />
    <path
      clipRule="evenodd"
      d="M3.26622 19.1615C3.62672 18.79 4.21764 18.7835 4.58608 19.1471C5.77978 20.3249 7.25115 21.1956 8.87067 21.6804C10.4902 22.1651 12.2065 22.2484 13.8669 21.9225C15.5272 21.5966 17.0788 20.8721 18.3845 19.8147C19.6902 18.7575 20.7095 17.4005 21.354 15.8651C21.9985 14.3299 22.249 12.6625 22.0841 11.0103C21.9191 9.35803 21.3437 7.76994 20.407 6.38681C19.4703 5.00356 18.2008 3.86748 16.7095 3.08099C15.2181 2.29444 13.5519 1.88235 11.8586 1.88235C11.3432 1.88235 10.9253 1.46097 10.9253 0.941176C10.9253 0.421379 11.3432 -4.54098e-07 11.8586 -4.76629e-07C13.8517 -5.63749e-07 15.8149 0.484934 17.5746 1.41299C19.3344 2.34109 20.8372 3.68426 21.9485 5.32528C23.0599 6.96643 23.7449 8.8543 23.9413 10.8218C24.1377 12.7893 23.8392 14.7738 23.073 16.5989C22.3068 18.424 21.0972 20.0322 19.5533 21.2824C18.0095 22.5326 16.1787 23.3864 14.2235 23.7702C12.2684 24.1539 10.2476 24.056 8.33952 23.4849C6.43142 22.9138 4.69353 21.8866 3.28059 20.4925C2.91215 20.129 2.90572 19.5331 3.26622 19.1615Z"
      fill="#007079"
      fillRule="evenodd"
    />
  </svg>
  <span>
    text
  </span>
</div>
`;
